Source/DafnyCore/Auditor/Auditor.cs

```diff
@@ -0,0 +1,135 @@
+#nullable enable
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.Linq;
+
+namespace Microsoft.Dafny.Auditor;
+
+/// <summary>
+/// A rewriter pass that produces an AuditReport and writes it to either
+/// standard output or a file, in one of several formats. If neither a file
+/// or a format is provided, it reports all assumptions as warnings.
+/// </summary>
+public class Auditor : IRewriter {
+  public enum ReportFormat { HTML, MarkdownTable, MarkdownIETF, Text }
+
+  public static readonly Option<string> ReportFileOption = new("--report-file",
+    "Specify a path to store the audit report file. Without this, the report will take the form of standard warnings.");
+  public static readonly Option<bool> CompareReportOption = new("--compare-report",
+    "Compare the report that would have been generated with the existing file given by --report-file, and fail if they differ.");
+
+  public static string SupportedReportFormats = "plain text in the format of warnings ('txt', 'text'); standalone HTML ('html'); a Markdown table ('md', 'markdown', 'md-table', 'markdown-table'); or an IETF-language document in Markdown format ('md-ietf', 'markdown-ietf')";
+
+  public static readonly Option<ReportFormat?> ReportFormatOption = new("--report-format",
```

Comment:
For the format, I guess I could argue either way. For the file, I think just using `--file` would be confusing, because it could refer to an input (`.dfy`) file, too.